### PR TITLE
[CIRCLE-14052]: Add `orb info` which reports meta-data about an orb

### PR DESCRIPTION
**Pull request by zzak**
## Help text

```
$ go run main.go help orb info
Show the meta-data of an orb


Usage:
  circleci orb info <orb> [flags]

Examples:
  circleci orb info circleci/python@0.1.4
  circleci orb info my-ns/foo-orb@dev:latest
```

## Usage

```
$ go run main.go orb info circleci/orb-tools

Latest: circleci/orb-tools@2.0.0
Last-updated: 2018-10-11T22:12:19.477Z
Created: 2018-09-24T07:55:19.386Z
First-release: 0.0.1 @ 2018-09-24T08:53:37.086Z
Total-revisions: 3

Total-commands: 4
Total-executors: 0
Total-jobs: 3
```

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -819,6 +829,52 @@ func OrbSource(ctx context.Context, log *logger.Logger, cl *client.Client, orbRe
 	return response.OrbVersion.Source, nil
 }
 
+// OrbInfo gets the meta-data of an orb
+func OrbInfo(ctx context.Context, log *logger.Logger, cl *client.Client, orbRef string) (*OrbVersionResponse, error) {
```
Comment (eric-hu) [inline api/api.go]: How come there's both `cl` and a `client` variables in this function?

File: cmd/orb.go [hunk 2]
```diff
@@ -506,3 +527,68 @@ func showSource(opts orbOptions) error {
...
+	opts.log.Info("\n")
+
+	var (
+		jobs      = 0
+		commands  = 0
+		executors = 0
+	)
+
+	var raw map[string]interface{}
+	if err := yaml.Unmarshal([]byte(info.OrbVersion.Source), &raw); err != nil {
```
Comment (eric-hu) [inline cmd/orb.go]: What's the motivation for using this decoding scheme for orb source versus the one in `api.go`?

File: api/api.go (hunk 1 above)
Comment (zzak) [reply api/api.go]: `cl` here refers to the `client.Client` pointer which is passed in as an argument to this function.

`client` refers to the "client" package found here:
https://github.com/CircleCI-Public/circleci-cli/blob/master/client/client.go

Happy for naming suggestions.

**Comment by zzak** [comment]
@CircleCI-Public/dx-internal Merging this, happy for post-review comments

File: cmd/cmd_suite_test.go [hunk 3]
```diff
@@ -22,6 +22,33 @@ var _ = BeforeSuite(func() {
 	Ω(err).ShouldNot(HaveOccurred())
 })
 
+func withTempHomeConfig() (string, *os.File) {
```
Comment (eric-hu) [inline cmd/cmd_suite_test.go]: 👏

File: cmd/orb.go [hunk 4]
```diff
@@ -464,45 +491,94 @@ func promoteOrb(opts orbOptions) error {
...
+
+	log.Info("\n")
+
+	var (
+		jobs      = 0
+		commands  = 0
+		executors = 0
+	)
+
+	for range info.Orb.Commands {
```
Comment (eric-hu) [inline cmd/orb.go]: `Orb.Commands` is a map, I think you can use `len(info.Orb.Commands)`? (Ditto for executors and jobs).

Comment (zzak) [reply cmd/orb.go]: @eric-hu Thanks for the feedback! I've added some of these changes to #191 :bow:

**Pull request merged**